fix laz support (memory issue and lack in lazperf type)

pblottiere opened the pull request:
This PR fixes three issues in laz support:
- the lack of support for PC_INT64, PC_UINT64, PC_DOUBLE, and PC_FLOAT for laz compression
- a memory error due to a too soon free in the lifecycle of a PCPOINTLIST coming from a lazperf patch
- a pgsql expected output for laz tests

mbredif commented on lib/pc_patch_lazperf.c:
```diff
@@ -70,10 +70,22 @@ pc_patch_lazperf_from_uncompressed(const PCPATCH_UNCOMPRESSED *pa)
-	pu = pc_patch_uncompressed_from_lazperf(palaz);
-	PCPOINTLIST *pl = pc_pointlist_from_uncompressed(pu);
-	pc_patch_free((PCPATCH *)pu);
+	PCPATCH_UNCOMPRESSED *pu = pc_patch_uncompressed_from_lazperf(palaz);
```
That looks good.
Maybe this memory issue fix could be factored out by creating a `pc_pointlist_clone` function to facilitate code reuse and unit testing.

elemoine replied:
Yes, I think `pc_pointlist_from_ght` suffers the same problem.

mbredif commented:
@pblottiere, #136 fixed the memory issue. Could you please rebase this PR to address the 2 other items only?

elemoine commented:
> @pblottiere, #136 fixed the memory issue. Could you please rebase this PR to address the 2 other items only?

Let me do it.

elemoine commented:
Done. @mbredif please confirm that this is what you expected.

mbredif commented:
Yes that is what I meant. @pblottiere do you confirm this simplified PR?

pblottiere commented:
Yes. Thank you @mbredif and @elemoine for taking in charge this PR!

elemoine commented:
@strk this was reviewed by @mbredif and @pblottiere. Could you please merge this?

pblottiere commented:
I rebased and added unit and integration test to confirm the behaviour whith PC_INT64, PC_UINT64, PC_DOUBLE, and PC_FLOAT dimensions during the lazperf compression.

elemoine commented on lib/cunit/cu_pc_patch_lazperf.c:
```diff
@@ -281,6 +289,47 @@ test_patch_filter_lazperf_zero_point()
+		pt = pc_point_make(multipledimschema);
+		pc_point_set_double_by_name(pt, "x", i*2);
+		pc_point_set_double_by_name(pt, "y", i*1.9);
+		pc_point_set_double_by_name(pt, "Z", i*0.34);
```
Minor, but is there any reason for using a capital Z and lower-case for x and y?

pblottiere commented:
I took into account your comments.

mbredif commented:
As a note, the `make installcheck` fails when laz is not installed, mostly due to [compression tests using laz within sql/pointcloud.sql](https://github.com/pblottiere/pointcloud/blob/abce712ec8171ab47118ab8c3ad9e6f22cfc093d/pgsql/sql/pointcloud.sql#L322). While your are modifying the expected output of this test, could this be moved to pgsql/sql/pointcloud-laz.sql ?

elemoine commented:
> As a note, the make installcheck fails when laz is not installed, mostly due to compression tests using laz within sql/pointcloud.sql. While your are modifying the expected output of this test, could this be moved to pgsql/sql/pointcloud-laz.sql ?

This PR is about supporting other dimension types for the lazperf compression. So I'd suggest that we work on separate PRs for this issue.

elemoine commented:
This looks good to go to me.

elemoine commented:
Great work @pblottiere!
